src/java.base/share/classes/sun/security/x509/CertificatePoliciesExtension.java

```diff
@@ -106,10 +106,14 @@ public CertificatePoliciesExtension(List<PolicyInformation> certPolicies)
      * a List of PolicyInformation with specified criticality.
      *
      * @param critical true if the extension is to be treated as critical.
-     * @param certPolicies the List of PolicyInformation.
+     * @param certPolicies the List of PolicyInformation, cannot be null or empty.
      */
     public CertificatePoliciesExtension(Boolean critical,
             List<PolicyInformation> certPolicies) throws IOException {
+        if (certPolicies == null || certPolicies.isEmpty()) {
```

Comment:
You should probably also change `PolicyInformation(CertificatePolicyId policyIdentifier,            Set<PolicyQualifierInfo> policyQualifiers)` to check for a null policyIdentifier and an empty policyQualifiers set.